lib/gateway/ShardManager.ts

```diff
@@ -24,16 +25,16 @@ try {
 /** A manager for all the client's shards. */
 export default class ShardManager extends Collection<number, Shard> {
     private _buckets: Record<number, number>;
-    private _client: Client;
     private _connectQueue: Array<Shard>;
     private _connectTimeout: NodeJS.Timeout | null;
     private _gatewayURL?: string;
+    client!: Client;
     connected = false;
+    dispatcher!: Dispatcher;
     options: ShardManagerInstanceOptions;
     constructor(client: Client, options: GatewayOptions = {}) {
         super();
         this._buckets = {};
-        this._client = client;
```

Comment:
you don't assign `this.client` anymore?

this seems to make the subsequent `this.client` calls in the constructor throw type errors. for example, giving the bot an unknown intent will trigger `this.client.emit("warn", ...)` at line 104, which will systematically throw an error, because `this.client` isn't defined. idk if this is intentional or not (i see you added a non-null assertion at property definition)

Reply:
See the defineProperties code at the bottom of the constructor

Reply:
oh you're right, missed that sry, so I guess changing line 104 by using the `client` parameter directly should do the trick